src/sonic_ax_impl/lib/quaggaclient.py

```diff
@@ -133,11 +134,22 @@ def vtysh_run(self, command):
     def vtysh_recv(self):
         acc = b""
         while True:
-            data = self.sock.recv(1024)
+            try:
+                data = self.sock.recv(1024)
+            except socket.timeout as e:
+                raise ValueError('Timeout recv acc=: {0}'.format(acc)) from e
             if not data:
-                raise ValueError('Unexpected data recv: {0}'.format(acc))
+                raise ValueError('Unexpected data recv acc=: {0}'.format(acc))
             acc += data
-            if acc.endswith(self.prompt_hostname):
+            ## 1. To match hostname
+            ##    RFC 1123 Section 2.1
+            ##    Hostname starts with with alphabet or number
```

Comment:
Suggest to change to `First char of hostname must be letter or number`

Reply:
Resolve, I even change more words

---
In reply to: [389239624](https://github.com/Azure/sonic-snmpagent/pull/124#discussion_r389239624) [](ancestors = 389239624)